integrate/simpsons_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import (
+	"math"
+	"testing"
+
+	"gonum.org/v1/gonum/floats"
+)
+
+func TestSimpsons(t *testing.T) {
+	const N = 1e6
+	x := floats.Span(make([]float64, N), 0, 1)
+	for i, test := range []struct {
+		x    []float64
+		f    func(x float64) float64
+		want float64
+	}{
+		{
+			x: []float64{0.0, 0.1, 1.0},
+			f: func(x float64) float64 {
+				switch {
+				case x >= 1e-31 || x <= 1.0:
```

Comment:
What is this test case testing? I think you wanted to put here `&&` instead of `||`. When you fix it, the test will fail.

Reply:
Also on this, please use the convention that intervals are written in number line order, this will be (with @vladimir-ch's correction) as `1e-31 <= x && x <= 1`. This eases identification of this kind of error.

Reply:
And in case my message was not clear, the test will fail as it should because this is a bad test case.

Reply:
I'm using a piecewise function.

Reply:
I had been looking at the stats package. I was hoping to approximate functions having more points at an endpoint or middle given a piecewise functions that plots points along a polynomial and a distribution chosen at random. I'd like to leave this bit to someone else if possible.

I still need to change the interval.

Reply:
It's not a piecewise constant function unless you put `&&` instead of `||` here.

Apart from that, it's still a bad test case. You are supplying a discontinuous function to a method that assumes smooth input. And you are sampling this function at only three locations, two fall into the 1 region, one falls into the 0 region. `Simpsons` integrates the quadratic function that interpolates these three values and returns the result which has nothing to the with the original piecewise constant function.